src/utils/taskUtils.ts

```diff
@@ -0,0 +1,53 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { Task, tasks as codeTasks, TaskScope, WorkspaceFolder } from "vscode";
+import { isPathEqual, isSubpath } from "./fs";
+
+export namespace taskUtils {
+    export function getFsPathFromTask(task: Task): string | undefined {
+        if (typeof task.scope === 'object') {
+            const workspaceFolder: Partial<WorkspaceFolder> = task.scope;
+            return workspaceFolder.uri?.fsPath;
+        } else {
+            return undefined;
+        }
+    }
+
+    /**
+     * Returns true if the task's scope is a workspace folder matching the given path or if the task is not associated with a path
+     */
+    export function isTaskInScopeOfPath(task: Task, fsPath: string): boolean {
```

Comment:
Is it worth removing functions in util files that aren't currently being used?

Reply:
Mmm could go either way. In general, I think it's okay to give "utils" files a bit of lee-way with our normal rules. For example, we allow a folder "utils" that doesn't have a very descriptive name and we frequently just copy code from other repos